### ISSUE TITLE
Doesn't work with mingw64 clang (edit: fixed)

### ISSUE THREAD

**Issue by rgkirch** [carp-lang/Carp#1323]
It doesn't work if I use clang from mingw but it works with clang from scoop. I got the idea to try installing with scoop because that's what the github action for windows uses. I ran the same command `scoop install llvm` and then just reordered the windows path.

```
$ echo %CARP_DIR%
C:\Users\richie\carp-v0.5.3-x86_64-windows
```

```
$ where clang
C:\msys64\mingw64\bin\clang.exe
C:\Users\richie\scoop\shims\clang.exe

$ carp handmade.carp -x
clang-cl: warning: unable to find a Visual Studio installation; try running Clang from a developer command prompt [-Wmsvc-not-found]
out\main.c(1,10): fatal error: 'assert.h' file not found
#include <assert.h>
         ^~~~~~~~~~
1 error generated.
carp: callCommand: clang-cl.exe  -o out\Untitled -I C:\Users\richie\carp-v0.5.3-x86_64-windows/core/  -D_CRT_SECURE_NO_WARNINGS out\main.c (exit 1): failed

$ clang --version
clang version 12.0.1
Target: x86_64-w64-windows-gnu
Thread model: posix
InstalledDir: C:/msys64/mingw64/bin

$ carp
> native-triple
=> ["x86_64" "mingw32" "unknown"]
```

`C:\msys64\clang64\bin\clang.exe` which I neglected to test until now gives the same error output above. This is the one that I thought I was using until I started comparing clangs. I was surprised to find `C:\msys64\mingw64\bin\clang.exe` 

I'm almost sure `msys64/clang64` came from installing `mingw-w64-clang-x86_64-clang`.

In total I've installed 3 things but idk what the difference is between `clang-x86_64-clang` and `x86_64-clang`

`pacman -S mingw-w64-clang-x86_64-clang mingw-w64-x86_64-clang mingw-w64-i686-clang`

```
$ where clang
C:\Users\richie\scoop\shims\clang.exe
C:\msys64\mingw64\bin\clang.exe

$ carp handmade.carp -x
hello world

$ clang --version
clang version 12.0.1
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Users\richie\scoop\apps\llvm\current\bin

$carp
> native-triple
=> ["x86_64" "mingw32" "unknown"]
```

This inspired me to search "clang scoop vs mingw" and I found an [SO answer](https://stackoverflow.com/a/42256528) about using `vcvarsall.bat` to setup the headers for clang.

```
$ where clang
C:\msys64\clang64\bin\clang.exe
C:\msys64\mingw64\bin\clang.exe
C:\Users\richie\scoop\shims\clang.exe

$ carp handmade.carp -x
clang-cl: warning: unable to find a Visual Studio installation; try running Clang from a developer command prompt [-Wmsvc-not-found]
out\main.c(1,10): fatal error: 'assert.h' file not found
#include <assert.h>
         ^~~~~~~~~~
1 error generated.
carp: callCommand: clang-cl.exe  -o out\Untitled -I C:\Users\richie\carp-v0.5.3-x86_64-windows/core/  -D_CRT_SECURE_NO_WARNINGS out\main.c (exit 1): failed

$ "c:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" x64
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.4
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

$ carp handmade.carp -x
hello world

$ "c:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat" amd64
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.4
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

$ carp handmade.carp -x
hello world
```

Now that I know to look for it, searching Carp issues for `vcvarsall` brings up https://github.com/carp-lang/Carp/issues/700

I don't have an issue anymore but I want to at least record this. I'd like to enhance the docs but I'd like to first know which mingw packages are actually needed. I'd like to at least see a mention of `vcvarsall.bat`.

```
$ pacman -Ss mingw-w64-clang-x86_64-clang
clang64/mingw-w64-clang-x86_64-clang 12.0.1-5 (mingw-w64-clang-x86_64-toolchain)
    C language family frontend for LLVM (mingw-w64)
```
```
$ pacman -Ss mingw-w64-x86_64-clang
mingw64/mingw-w64-x86_64-clang 12.0.1-5
    C language family frontend for LLVM (mingw-w64)
```

Does the `mingw-w64-clang-x86_64-clang` include a bunch of packages and `mingw-w64-x86_64-clang` is one of the included ones?

https://packages.msys2.org/group/mingw-w64-clang-x86_64-toolchain

**Comment by rgkirch** [comment]
tldr:
I fixed my own problem by running `vcvarsall.bat` but I want the `Install.md` docs to mention it since it took me 2 days to figure out.

**Comment by rgkirch** [comment]
according to [SO](https://stackoverflow.com/a/1254011)

32/64 OS building for 32 => x86
32 OS building for 64 => x86_amd64
64 OS building for 64 => amd64
